ui/src/treeDataProvider/AzureTreeDataProvider.ts

```diff
@@ -81,49 +86,75 @@ export class AzureTreeDataProvider implements TreeDataProvider<IAzureNode>, Disp
-                nodes = [new AzureNode(undefined, {
-                    label: localize('loadingNode', 'Loading...'),
-                    commandId: loginCommandId,
-                    contextValue: 'azureCommandNode',
-                    id: loginCommandId,
-                    iconPath: {
-                        light: path.join(__filename, '..', '..', '..', '..', 'resources', 'light', 'Loading.svg'),
-                        dark: path.join(__filename, '..', '..', '..', '..', 'resources', 'dark', 'Loading.svg')
-                    }
-                })];
-            } else if (this._azureAccount.status === 'LoggedOut') {
-                nodes = [
-                    new AzureNode(undefined, { label: localize('signInNode', 'Sign in to Azure...'), commandId: loginCommandId, contextValue: 'azureCommandNode', id: loginCommandId }),
-                    new AzureNode(undefined, { label: localize('createNode', 'Create a Free Azure Account...'), commandId: createCommandId, contextValue: 'azureCommandNode', id: createCommandId })
-                ];
-            } else if (this._azureAccount.filters.length === 0) {
-                commandLabel = localize('noSubscriptionsNode', 'No subscriptions found. Edit filters...');
-                nodes = [new AzureNode(undefined, { label: commandLabel, commandId: 'azure-account.selectSubscriptions', contextValue: 'azureCommandNode', id: 'azure-account.selectSubscriptions' })];
-            } else {
-                this._subscriptionNodes = this._azureAccount.filters.map((subscriptionInfo: AzureSubscription) => {
-                    if (subscriptionInfo.subscription.subscriptionId === undefined || subscriptionInfo.subscription.displayName === undefined) {
-                        throw new ArgumentError(subscriptionInfo);
+        try {
+            // tslint:disable:no-var-self
+            const thisTree: AzureTreeDataProvider = this;
+            return <IAzureNode[]>await callWithTelemetryAndErrorHandling('AzureTreeDataProvider.getChildren', this._telemetryReporter, undefined, async function (this: IActionContext): Promise<IAzureNode[]> {
+                const actionContext: IActionContext = this;
+                // tslint:enable:no-var-self
+                actionContext.suppressErrorDisplay = true;
+                actionContext.rethrowError = true;
+                let result: IAzureNode[];
+
+                if (node !== undefined) {
+                    actionContext.properties.contextValue = node.treeItem.contextValue;
+
+                    result = node.creatingNodes
+                        .concat(await node.getCachedChildren())
+                        .concat(node.treeItem.hasMoreChildren() ? new AzureNode(node, new LoadMoreTreeItem(thisTree._loadMoreCommandId)) : []);
+                } else { // Root of tree
+                    actionContext.properties.isActivationEvent = 'true';
```

Comment:
What does isActivationEvent mean again?

Reply:
It means that this event happens as a part of extension activation - aka the event is sent even if they don't interact with the explorer or our extension at all (which we shouldn't be counting as 'usage' of our extension).

In this case, it means I don't want to count 'getChildren' at the root of the explorer as 'usage' for our extension